tests/integration/pilot/common/routing.go

```diff
@@ -452,6 +452,44 @@ spec:
 	return []TrafficTestCase{tc}
 }
 
+// useClientProtocolCases contains tests use_client_protocol from DestinationRule
+func useClientProtocolCases(apps *EchoDeployments) []TrafficTestCase {
+	var cases []TrafficTestCase
+	client := apps.PodA
+	destination := apps.PodC[0]
+	cases = append(cases,
+		TrafficTestCase{
+			name:   "use client protocol with h2",
+			skip:   apps.IsMulticluster(), // Not needed for multi cluster.
+			config: useClientProtocolDestinationRule("use-client-protocol-h1", destination.Config().Service),
+			call:   client[0].CallWithRetryOrFail,
+			opts: echo.CallOptions{
+				Target:   destination,
+				PortName: "http",
+				HTTP2:    true,
+				Validator: echo.And(
+					echo.ExpectOK(),
+				),
+			},
+		},
+		TrafficTestCase{
+			name:   "use client protocol with h1",
+			skip:   apps.IsMulticluster(), // Not needed for multi cluster.
+			config: useClientProtocolDestinationRule("use-client-protocol-h1", destination.Config().Service),
+			call:   client[0].CallWithRetryOrFail,
+			opts: echo.CallOptions{
+				PortName: "http",
+				Target:   destination,
+				HTTP2:    false,
```

Comment:
It seems like this doesn't actually test that we preserve the client protocol. We check that we can send h1 and send h2, but we need to check that when we send h1 the backend gets h1. It may require changes to the echo app to emit what protocol was used to hit it?

Reply:
Yeah.. I wasn't sure how to do it. Any suggestions?

Reply:
`echo.ExpectKey("Proto", "HTTP/2.0")` and `echo.ExpectKey("Proto", "HTTP/1.1")`  I think.